src/util/ref.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_REF_H
+#define BITCOIN_UTIL_REF_H
+
+#include <util/check.h>
+
+#include <typeindex>
+
+namespace util {
+
+/**
+ * Type-safe dynamic reference.
+ *
+ * This implements a small subset of the functionality in C++17's std::any
```

Comment:
Thanks for the review. std::any can hold pointers and reference wrappers, so there is no problem here. I'm going to mark the style comments resolved because I disagree with them and don't want to create extended off-topic conversations. But if there's anything that should be reconsidered, definitely do follow up. Review suggestions tend to be most useful when they are specific and say something about the motivation behind the suggestion. That way, if there's disagreement about the suggestion, something else can be done to address whatever the motivation for it was.